utils/src/main/java/com/revenuecat/purchases/utils/Result.kt

```diff
@@ -0,0 +1,6 @@
+package com.revenuecat.purchases.utils
+
+sealed class Result<out A, out B> {
+    class Success<A>(val value: A): Result<A, Nothing>()
+    class Error<B>(val value: B): Result<Nothing, B>()
+}
```

Comment:
This is inspired from `purchases-ios` and `https://github.com/arrow-kt/arrow/blob/main/arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt`. I feel this can be useful, especially if we move towards coroutines in the future but lmk if you prefer not to introduce something new now and I can remove it.

Reply:
Love this. We should have introduced this class long time ago. I used to have the same in other projects